src/qibo/backends/__init__.py

```diff
@@ -117,37 +116,58 @@ def set_backend(cls, backend, **kwargs):
     @classmethod
     def transpiler(cls):
         """Get the current transpiler. If no transpiler is set, it will create one."""
+        if cls._transpiler is not None:
+            return cls._transpiler
+
+        cls._transpiler = cls._default_transpiler()
+        return cls._transpiler
+
+    @classmethod
+    def set_transpiler(cls, transpiler):
+        cls._transpiler = transpiler
+        # TODO: check if transpiler is valid on the backend
+
+    @classmethod
+    def _default_transpiler(cls):
+        import networkx as nx
+
         from qibo.transpiler.optimizer import Preprocessing
         from qibo.transpiler.pipeline import Passes
         from qibo.transpiler.placer import Trivial
         from qibo.transpiler.router import Sabre
-        from qibo.transpiler.unroller import Unroller
+        from qibo.transpiler.unroller import NativeGates, Unroller
 
-        if cls._transpiler is not None:
-            return cls._transpiler
+        qubits: list[str] = cls._backend.qubits  # list of qubit names
+        natives: list[str] = cls._backend.natives
+        connectivity_edges: list[tuple[str, str]] = (
+            cls._backend.connectivity
+        )  # list of edges
+        if qubits and natives and connectivity:
+
+            # code is needed in unroll.py to convert str -> enum
+            natives_enum = NativeGates.from_gatelist(natives)
 
-        if cls._backend.name == "qibolab":  # pragma: no cover
-            platform = cls._backend.platform
-            natives = platform.natives  # qibolab 0.2.0
-            connectivity = platform.topology  # qibolab 0.1.9
+            connectivity = nx.Graph()
+
+            # q{i} naming
+            node_mapping = {qubits[i]: i for i in range(len(qubits))}
+            for e in connectivity_edges:
+                connectivity.add_edge(node_mapping[e[0]], node_mapping[e[1]])
```

Comment:
We could this in a different PR (in which case, open a dedicated issue), but I would go the other way round: use integers in the transpiler, do not convert them to strings.

So: if a `Circuit` has no `wire_names` set (which may be a bit of a challenge to find out, but we should be able to), you just use the qubits' position, as it is right now, otherwise
- convert `wire_names` to indices in the platform listing
- permute the `Circuit` in a first step, and pad it properly